include/picolibrary/fixed_size_array.h

```diff
@@ -0,0 +1,307 @@
+/**
+ * picolibrary
+ *
+ * Copyright 2020 Andrew Countryman <apcountryman@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/**
+ * \file
+ * \brief picolibrary::Fixed_Size_Array interface.
+ */
+
+#ifndef PICOLIBRARY_FIXED_SIZE_ARRAY_H
+#define PICOLIBRARY_FIXED_SIZE_ARRAY_H
+
+#include <cstddef>
+
+namespace picolibrary {
+
+/**
+ * \brief Fixed size array.
+ *
+ * \tparam T The array element type.
+ * \tparam N The number of elements in the array.
+ */
+template<typename T, std::size_t N>
+class Fixed_Size_Array {
+  public:
+    /**
+     * \brief The array element type.
+     */
+    using Value = T;
+
+    /**
+     * \brief The number of elements in the array.
+     */
+    using Size = std::size_t;
+
+    /**
+     * \brief An array element position.
+     */
+    using Position = std::size_t;
+
+    /**
+     * \brief A reference to an array element.
+     */
+    using Reference = Value &;
+
+    /**
+     * \brief A reference to a const array element.
+     */
+    using Const_Reference = Value const &;
+
+    /**
+     * \brief A pointer to an array element.
+     */
+    using Pointer = Value *;
+
+    /**
+     * \brief A pointer to a const array element.
+     */
+    using Const_Pointer = Value const *;
+
+    /**
+     * \brief An array iterator.
+     */
+    using Iterator = Pointer;
+
+    /**
+     * \brief A const array iterator.
+     */
+    using Const_Iterator = Const_Pointer;
+
+    /**
+     * \brief The underlying array.
+     */
+    Value m_array[ N ];
+
+    /**
+     * \brief Access the element at the specified position in the array.
+     *
+     * \param[in] position The position of the array element to access.
+     *
+     * \warning Bounds checking is not performed.
+     *
+     * \warning Calling this function on an empty array results in undefined behavior.
```

Comment:
Remove blank line between warnings and move them to before the parameter documentation.